components/admin/product/productFetch/productFetch.tsx

```diff
@@ -41,19 +25,35 @@ import {
     TabsList,
     TabsTrigger,
 } from "@/components/ui/tabs"
-
 import StatusandFilter from "@/components/admin/product/productutils/layout/statusandFilter"
 import { DropDownMenu } from "@/components/admin/product/productutils/layout/dropDownMenu"
-import { ProductsContext, Productsprops } from '@/components/context/mycontext';
-import { useContext } from 'react';
+import { ProductsContext, Productsprops, productlistprop } from '@/components/context/mycontext';
 
 import { useState, useEffect } from "react";
-import {useStore,useToken} from "@/components/helpers/zustand"
-import ApiFetchProducts from "@/components/admin/product/productFunctions/apiFetchProducts"
-// import useJwtToken from "@/components/helpers/getToken"
+import { useToken } from "@/components/helpers/zustand"
+
+import { api } from "@/trpc/react"
+// import { LoadingPage, LoadingSpinner } from "@/components/admin/product/productutils/loaders";
+// import { Button } from "@/components/ui/button"
+import { Input } from "@/components/ui/input"
+import { Label } from "@/components/ui/label"
+import {
+    Sheet,
+    SheetClose,
+    SheetContent,
+    SheetDescription,
+    SheetFooter,
+    SheetHeader,
+    SheetTitle,
+    SheetTrigger,
+} from "@/components/ui/sheet"
+import ProductUpdate from "@/components/admin/product/productUpdate/productUpdate"
 export function Dashboard() {
-    
- const data = useStore((state)=>(state.data));
+    const token = useToken((state) => state.token);
+    const [fetchedData, setFetchedData] = useState<productlistprop>();
+    const [sheetOpen, setSheetOpen] = useState(false);
+    const [index, setIndex] = useState(0);
+    const { data, isLoading, refetch, error } = api.product.productfetch.useQuery( { jwtToken: token || '' } );
```

Comment:
here instead of manually passing token everytime... make a protectedProcedure(watch theo video for reference) in TRPC for API reoutes which require auth